src/GUI/EFCorePowerTools/ViewModels/ObjectTreeViewModel.cs

```diff
@@ -41,11 +42,20 @@ public class ObjectTreeViewModel : ViewModelBase, IObjectTreeViewModel
 
         public bool IsInEditMode { get => _objects.Any(o => o.IsEditing) || _objects.SelectMany(c => c.Columns).Any(o => o.IsEditing); }
 
-        public void Search(string searchText)
+        public void Search(string searchText, SearchMode searchMode)
         {
+            var regex = new Regex(searchText);
+
             foreach (var obj in _objects)
             {
-                obj.IsVisible = string.IsNullOrWhiteSpace(searchText) || obj.DisplayName.ToUpper().Contains(searchText.ToUpper());
+                if (searchMode == SearchMode.Text)
+                {
+                    obj.IsVisible = string.IsNullOrWhiteSpace(searchText) || obj.DisplayName.ToUpper().Contains(searchText.ToUpper());
+                }
+                else
+                {
+                    obj.IsVisible = regex.IsMatch(obj.DisplayName);
```

Comment:
Pls add a timeout.

Reply:
What do you mean by "timeout"? Something like: "if the search is not over within 3 seconds abort it?" or something else?

Reply:
Yes, it is a security issue to not add a timeout, say 5 seconds (via RegExOptions)

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices#use-time-out-values

Reply:
This days isn't useless, I've learnt something new :D